src/input-screen/index.js

```diff
@@ -81,45 +85,60 @@ class FileSelection extends PureComponent {
 		// Only accept files if one isn't already selected
 		const DropZoneTarget = file ? Fragment : DropZoneProvider;
 
+		const hasImportableResource = file || url.length; // We would want to bolster the URL validation eventually
+
 		return (
 			<Fragment>
-				<DropZoneTarget>
-					<h2>Import WordPress</h2>
-					<div>Howdy! Upload your WordPress eXtended RSS (WXT) file and we'll import the posts, pages, comments, custom fields, categories, and tags into this site.</div>
-					<div>Choose a WXR (.xml) file to upload, or drop a file here, and your import will begin</div>
-					{ ! url && (
-						<div className="wordpress-importer__div-actions">
-							Import from file:
-							{ file
-								? ( <span>{ file.name } <Button onClick={ () => this.setState( { file: null } ) } isLink><Icon icon="no" /></Button></span> )
-								: ( <FileInput onFileSelected={ this.handleFileSelection }>Choose file</FileInput> )
-							}
+				<h2>Import WordPress</h2>
+				<div>Howdy! Upload your WordPress eXtended RSS (WXT) file and we'll import the posts, pages, comments, custom fields, categories, and tags into this site.</div>
+				<div>Choose a WXR (.xml) file to upload, or drop a file here, and your import will begin</div>
+				<div className="wordpress-importer__div-actions">
+					{ file ? (
+						<div className="wordpress-importer__file-select-action">
+							<span>
+								{ file.name }
+								<Button
+									onClick={ () => this.setState( { file: null } ) }
+									isLink
+								>
+									<Icon icon="no" />
+								</Button>
+							</span>
 						</div>
-					) }
-					{ ! file && (
-						<div className="wordpress-importer__div-actions">
-							<TextControl
-								label="Import from url:"
-								onChange={ ( url ) => this.setState( { url } ) }
-								value={ url }
+					) : (
+						<DropZoneTarget>
+							<div className="wordpress-importer__url-input-action">
+								<TextControl
+									label="Import from url:"
+									onChange={ ( url ) => this.setState( { url } ) }
+									value={ url }
+								/>
+							</div>
+							<div className="wordpress-importer__file-select-action">
+								<FileInput
+									onFileSelected={ this.handleFileSelection }
+								>
+									Or drag a file here, or click to select.
```

Comment:
How about "You can also drag a file here, or click to browse."?

Reply:
I like it :) gets rid of that double 'or' quite nicely! Update in [81a00b8](/Automattic/wordpress-importer/pull/12/commits/81a00b8e237e216e1392b9e477b357a613979d1a) 👌